core/src/scene/drawRule.cpp

```diff
@@ -48,6 +48,19 @@ DrawRule::DrawRule(const DrawRuleData& _ruleData, const SceneLayer& _layer) :
     }
 }
 
+DrawRule::DrawRule(const DrawRule& _other) : active(_other.active), name(_other.name),
```

Comment:
I believe we can use the default copy constructor. I see that you took care to only copy the "active" parameters, but we can safely copy all of the parameters (which are just pointers) since the "active" bits are copied too.